### PR TITLE
Destroy callback group before node

### PR DESCRIPTION
**Pull request by mjcarroll**
The new executor implementation gives an exception if callback groups and nodes are destructed out of order.  

It is incorrect for callback groups to be held alive longer than their owning nodes, but we previous didn't warn about it.

### CODE REVIEW

**Review by clalancette — APPROVED** [review]
Yeah, this makes sense to me.  Will run CI on just this change next.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18526/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13073/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19256/)

File: tf2_ros/include/tf2_ros/transform_listener.h [hunk 1]
```diff
@@ -243,6 +242,8 @@ class TransformListener
   tf2::TimePoint last_update_;
   rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging_interface_ {nullptr};
   rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base_interface_ {nullptr};
+  rclcpp::CallbackGroup::SharedPtr callback_group_{nullptr};
+
```
Comment (clalancette) [inline tf2_ros/include/tf2_ros/transform_listener.h]: Uncrustify is not happy with this extra line.

```suggestion
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18531/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13077/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19262/)

**Review by clalancette — APPROVED** [review]

**Pull request merged**